apps/glusterfs/brick_allocate.go

```diff
@@ -16,17 +16,51 @@ import (
 	"github.com/heketi/heketi/pkg/utils"
 )
 
-type BrickAllocation struct {
+type BrickSet struct {
+	SetSize int
 	Bricks  []*BrickEntry
+}
+
+func NewBrickSet(s int) *BrickSet {
+	return &BrickSet{SetSize: s, Bricks: []*BrickEntry{}}
+}
+
+func (bs *BrickSet) Add(b *BrickEntry) {
+	bs.Bricks = append(bs.Bricks, b)
+}
+
+func (bs *BrickSet) Full() bool {
```

Comment:
The `Full()` methods are not used in this PR. Would we want to use them in `Add()` and prevent from over-adding? And maybe adding a (error) return value to `Add()`?